include/swift/Option/Options.td

```diff
@@ -331,6 +331,16 @@ def Rpass_missed_EQ : Joined<["-"], "Rpass-missed=">,
   HelpText<"Report missed transformations by optimization passes whose "
            "name matches the given POSIX regular expression">;
 
+def fsave_optimization_record : Flag<["-"], "fsave-optimization-record">,
```

Comment:
Swift doesn't use "-f" prefixes. Additionally, I'm not sure it makes sense to have separate options for "save optimization record" and "record optimization file".

(The Swift convention for this is `-emit-foo-bar` to emit something to its default location and `-emit-foo-bar-path <path>` to specify it explicitly. I can understand the desire to more closely parallel Clang, though.)

Reply:
Since we're removing the -f prefix we might as well make this follow the Swift convention. 

Changing this to -save-optimization-record and -save-optimization-record-path.

Reply:
@jrose-apple, I've looked into how .dia files are handled via addAuxiliaryOutput.  I am bit unclear how we want to handle WMO after that.

Do you want to handle the location for the single WMO YAML file via the output map or keep the ...-path option and only use that in the WMO mode?  (I don't want to add the ...-path option if we were going to remove it in the near future.)

Reply:
I didn't realize that serialized diagnostics didn't let you specify the file name without an output file map. I suppose I was thinking of something more like -emit-objc-header vs. -emit-objc-header-path (further down in Driver.cpp), where the latter lets you control the output file but the former just goes with the default path / the location specified by the output file map. The difference here is that there's only ever one ObjC header for an entire module, even in non-WMO mode.

But .d files (-emit-dependencies) behave the same as .dia files. Maybe that does make more sense as a model. In that case, you'd sink `-save-optimization-record-path` down to be a frontend-only option.